packages/core/src/Site/index.ts

```diff
@@ -401,66 +406,66 @@ export class Site {
     if (_.isUndefined(filePath)) return;
     try {
       await fs.copy(path.join(this.rootPath, filePath), indexPagePath);
+      logger.info(`Copied over the existing ${filePath} file to ${INDEX_MARKDOWN_FILE}`);
     } catch (error) {
       throw new Error(`Failed to copy over ${filePath}`);
     }
   }
 
   /**
-   * Adds an about page to site if not present.
+   * Creates and adds a default layout
+   * by rendering the specified nunjucks file with nunjucks variables.
+   *
+   * @param njkFile The nunjucks template to create a default layout from.
+   * @param {NunjuckObj=} [njkObject] The object containing the nunjucks variables of the template.
    */
-  async addAboutPage() {
-    const aboutPath = path.join(this.rootPath, ABOUT_MARKDOWN_FILE);
-    try {
-      await fs.access(aboutPath);
-    } catch (error) {
-      if (fs.existsSync(aboutPath)) {
-        return;
-      }
-      await fs.outputFile(aboutPath, ABOUT_MARKDOWN_DEFAULT);
-    }
+  addDefaultLayoutFile(njkFile: string, njkObject?: NunjuckObj) {
+    const convertedLayoutTemplate = VariableRenderer.compile(
+      fs.readFileSync(path.join(__dirname, njkFile), 'utf8'));
+    const renderedLayout = convertedLayoutTemplate.render(njkObject);
+    const layoutOutputPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH, LAYOUT_DEFAULT_NAME);
+
+    fs.writeFileSync(layoutOutputPath, renderedLayout, 'utf-8');
   }
 
   /**
-   * Adds a footer to default layout of site.
+   * Copies over the contents of substitution files and assigns them to njk variables.
+   *
+   * @param njkSubs The njk variables and their possible file substitutions.
+   * @param hasAutoSiteNav Creates navigation links if true.
+   * @param siteNavIgnore Files to ignore when creating navigation links.
+   * @returns A NunjuckObj with all nunjucks variables in the layout.
    */
-  addDefaultLayoutFiles() {
-    const wikiFooterPath = path.join(this.rootPath, WIKI_FOOTER_PATH);
-    let footer;
-    if (fs.existsSync(wikiFooterPath)) {
-      logger.info(`Copied over the existing ${WIKI_FOOTER_PATH} file to the converted layout`);
-      footer = `\n${fs.readFileSync(wikiFooterPath, 'utf8')}`;
-    }
+  createNjkObjects({ njkSubs, hasAutoSiteNav, siteNavIgnore }: TemplateConfig) {
+    const njkObject: NunjuckObj = {};
+    njkSubs.forEach((njkSub) => {
+      let fileCopy;
+      njkSub.fileSubstitutes.forEach((fileName) => {
+        const subFilePath = path.join(this.rootPath, fileName);
+        if (fs.existsSync(subFilePath)) {
+          logger.info(`Copied over the existing ${fileName} file to the converted layout`);
+          fileCopy = `\n${fs.readFileSync(subFilePath, 'utf8')}`;
```

Comment:
What I'm referring to in another comment: 
Only the last file is used as a substitute here. In that case, would be better to make it a string instead of an array of string